### PR TITLE
Structuring cli conf file and parser update with new clap macro

### LINKED ISSUE #242: Pyrsia CLI to have a default config
Right now once the cli is installed , user has to add config using "pyrsia config --add" to write to conf file , we can add a default config with default port, which needs initialization of config file at the time of installation

### PR DESCRIPTION
**Pull request by MitaliBo**
## PR Checklist

- [x] I've built the code `cargo build`. For major changes, `cargo build --workspace --release` is recommended.
- [x] I've run the automated unit tests `cargo test`. This executes our automated unit tests.
- [x] I've not broken any existing tests or functionality. In addition to `cargo test`, I've run `cargo clippy`
- [x] I've not introduced any new known security vulnerabilities. I've run `cargo audit`

- [x] I've included a brief description and a good title of the proposed changes and how to test them.
- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've associated an [issue](https://github.com/pyrsia/pyrsia/issues) with this Pull Request. 
- [x] I've checked that the code is up-to-date with the `pyrsia/main` branch.
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [x] I've assigned this Pull Request to "pyrsia/collaborators"

## Description

Fixes #242 

This PR does following 
1) Use of confy for pyrsia-cli configuration file which will have default config file and cli commands integrated to be able to 
    edit that file and test cases for same
2) Clap update to use updated macro (!command) and some cleanup for cli commands
3) update the same clap macro on pyrsia_node too

### CODE REVIEW

**Comment by prince-chrismc** [comment]
Test fail looks like a small typo, https://github.com/pyrsia/pyrsia/runs/5325711429?check_suite_focus=true#step:8:547

Only did a quick scan since I was looking into CI but seem like great work! 🚀

**Comment by MitaliBo** [comment]
> Test fail looks like a small typo, https://github.com/pyrsia/pyrsia/runs/5325711429?check_suite_focus=true#step:8:547
> 
> Only did a quick scan since I was looking into CI but seem like great work! 🚀

No Sadly that is not typo :( i was debugging why tests pass in my local but not in CI, let see i was able to identify correct issue and the fix works.

**Comment by MitaliBo** [comment]
@tiainen @erwin1 can you please review

File: pyrsia_cli/src/commands/config.rs [hunk 1]
```diff
@@ -15,30 +15,103 @@
...
+    }
+}
+
+pub fn add_config(new_cfg: CliConfig) -> Result<()> {
+    let mut cfg: CliConfig = confy::load(CONF_FILE)?;
+    if !new_cfg.host.is_empty() {
+        cfg.host = new_cfg.host
+    }
+
+    if !new_cfg.port.is_empty() {
```
Comment (tiainen) [inline pyrsia_cli/src/commands/config.rs:55]: Should the empty check also be done for the `disk_allocated` configuration?

**Review by erwin1 — DISMISSED** [review]
works fine!

File: pyrsia_cli/Cargo.toml [hunk 2]
```diff
@@ -19,4 +21,7 @@ name = "pyrsia"
 path = "src/main.rs"
 
 [dev-dependencies]
-expectest = "0.12.0"
+directories = "^2.0"
+assay = "0.1.0"
```
Comment (tiainen) [inline pyrsia_cli/Cargo.toml:25]: We recently updated assay to `0.1.1`.

Comment (prince-chrismc) [reply pyrsia_cli/Cargo.toml:25]: #426 👍

Comment (MitaliBo) [reply pyrsia_cli/Cargo.toml:25]: yes i only did, i think i have override my changes :)

**Review by erwin1 — APPROVED** [review]

**Review by sbtaylor15 — APPROVED** [review]
All the requested changes have been made and looks good.

**Pull request merged**